bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -330,5 +331,14 @@ std::string PrePrepareMsg::getCid() const {
   return std::string(this->body() + this->payloadShift() - b()->batchCidLength, b()->batchCidLength);
 }
 
+void PrePrepareMsg::updateCid(SeqNum s) {
+  std::string cidStr = std::to_string(s);
+  if (cidStr.size() >= b()->batchCidLength) {
+    memcpy(this->body() + this->payloadShift() - b()->batchCidLength, cidStr.data(), b()->batchCidLength);
+  } else {
+    memcpy(this->body() + this->payloadShift() - b()->batchCidLength + 1, cidStr.data(), b()->batchCidLength - 1);
```

Comment:
How this could happen now?

Reply:
As this is a new function, after adding this function, the cid will be populated again with the new seq number. So the cid and seq number will become same now for all PPMs after this change.